pypeit/core/arc.py

```diff
@@ -1107,7 +1107,11 @@ def fit_arcspec(xarray, yarray, pixt, fitp):
 #            continue  # Probably won't be a good solution
         # Fit the gaussian
         try:
-            pypeitFit = fitting.func_fit(xarray[pmin:pmax], yarray[pmin:pmax], "gaussian", 3)#, return_errors=True)
+            # TODO -- REPLACE THIS WITH A CUSTOM GAUSSIAN FITTER
+            pypeitFit = fitting.PypeItFit(xval=xarray[pmin:pmax], yval=yarray[pmin:pmax],
```

Comment:
It is ugly to have to cast the order to an array. Can we not just check internally to PypeItFit. The easiest way would be to cast inside of PypeItFit in which case order could be an integer, a 1d array, or a 2d array or tuple for 2d fits. I realize this breaks the automated code you have that sets the attributes of the datamodel, but is it really required that all arguments to PypeItFit are part of the datamodel?